src/vm/vm.c

```diff
@@ -2187,6 +2219,16 @@ static DictuInterpretResult run(DictuVM *vm) {
             ObjDict *dict = AS_DICT(READ_CONSTANT());
             ObjClass *klass = AS_CLASS(peek(vm, 0));
 
+            if (IS_CLASS(peek(vm, 1))) {
+                ObjClass *super = AS_CLASS(peek(vm, 1));
+
+                if (super->classAnnotations != NULL) {
+                    ObjDict *superAnnotations = copyDict(vm, super->classAnnotations, true);
+
+                    copyAnnotations(vm, superAnnotations, dict);
+                }
+            }
```

Comment:
I just kinda had a realisation with this bit of code as well (and the other two opcodes)

We can simply just do:
```suggestion
            if (klass->classAnnotations != NULL) {
                copyAnnotations(vm, klass->classAnnotations, dict);
            }
```

As the dict on the class has already been copied from the parent so we should be free to modify this directly